### PR TITLE
8295970: Add vector api sanity tests in tier1

### PR DESCRIPTION
**Pull request by DamonFool**
Hi all,

As discussed here https://github.com/openjdk/jdk/pull/10807#pullrequestreview-1150314487 , it would be better to add the vector api tests in GHA.

Thanks.
Best regards,
Jie

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295970](https://bugs.openjdk.org/browse/JDK-8295970): Add vector api sanity tests in tier1

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [f901b907](https://git.openjdk.org/jdk/pull/10879/files/f901b9072dd4659b9d4cb67d27793e1f47f20a1e)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [f901b907](https://git.openjdk.org/jdk/pull/10879/files/f901b9072dd4659b9d4cb67d27793e1f47f20a1e)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10879/head:pull/10879` \
`$ git checkout pull/10879`

Update a local copy of the PR: \
`$ git checkout pull/10879` \
`$ git pull https://git.openjdk.org/jdk pull/10879/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10879`

View PR using the GUI difftool: \
`$ git pr show -t 10879`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10879.diff">https://git.openjdk.org/jdk/pull/10879.diff</a>

</details>

### CODE REVIEW

**Review by shipilev — CHANGES_REQUESTED** [review]
Sorry, but I don't think this is a good idea. If there are tests that are important for pre-integration testing, they should just be in `tier1`.

**Comment by DamonFool** [comment]
> Sorry, but I don't think this is a good idea. If there are tests that are important for pre-integration testing, they should just be in `tier1`.

Since vector api is still in incubation, I'm not sure if it's OK to go into tier1.
I'm fine to add it in tier1 if there is no objection.
Thanks.

**Comment by mrserb** [comment]
Probably it will be possible to select an additional tests dynamically based on changed module? Using the same logic as the labels in the PR?

**Comment by DamonFool** [comment]
> Probably it will be possible to select an additional tests dynamically based on changed module? Using the same logic as the labels in the PR?

Thanks @mrserb for your suggestion.

Dynamical testing configuration would increase the GHA complexity, which seems not worth it for jdk_vector.
This is because all 3 tier1_part* and jdk_vector took about 5 min to run according to Oracle's internal testing times.

So I think it's fine to add it in GHA directly.
Thanks.

**Comment by erikj79** [comment]
The default set of tests included in GHA is currently defined to be exactly tier1. That is the set of tests that we have collectively agreed on represents a reasonably sized set that gives the best coverage for the time spent (at least that's the intention of tier1). The time spent here is probably the most important variable, as whatever we put here, everyone runs all the time in their PRs and PFs. Adding 5 minutes may not seem like much, but multiplied with platforms and all the users and PRs it adds up to a lot. Also imagine if more components would want to add to the set, pointing to this change as a precedent ,adding another 5 minutes, and then another 5 minutes, eventually the reasonable size of the test set becomes completely unreasonable.

My point is that we can't just add more and more to GHA on a whim. The current agreement is for it to be exactly tier1. If you can't motivate that your tests should be part of tier1, then I find it pretty hard to motivate that it should be part of GHA. I don't know enough about these tests to know if they should fit in tier1/GHA or not, I'm just pointing out that we need to keep a pretty high bar to avoid bloating the set.

For this particular issue, would it be possible to define a minimal set of vector api tests that would have discovered the problem in https://github.com/openjdk/jdk/pull/10807 without taking 5 minutes? Perhaps the better solution here would be to allow a single or very few specific tests into tier1?

**Comment by vnkozlov** [comment]
I think we can meet all requests by adding `jdk_vector_sanity` subset of vector tests similar to existing `jdk_svc_sanity`.
The motivation for that is to catch obvious bugs in vector code generation on 32-bit platforms and aarch64 very early.
We, in Oracle, help to test 64 bits but it is not enough I think.
Vector API is stable enough already and I think we can have some sanity checks for it in tier1.
The only issue is to find subset of tests.

**Comment by DamonFool** [comment]
> I think we can meet all requests by adding `jdk_vector_sanity` subset of vector tests similar to existing `jdk_svc_sanity`. The motivation for that is to catch obvious bugs in vector code generation on 32-bit platforms and aarch64 very early. We, in Oracle, help to test 64 bits but it is not enough I think. Vector API is stable enough already and I think we can have some sanity checks for it in tier1. The only issue is to find subset of tests.

Good suggestion!
And the `jdk_vector_sanity` test group had been added.

The subset is selected as in the following picture (red in the picture).
The XMaxVector tests are selected because they will be 512 vectors on x86/AVX512 and 256 vectors on x86/AVX2.
And for most cases, people would like to use the max length vectors.
Thanks.

image

**Comment by DamonFool** [comment]
> I think you need to add at least one other label than `build` to this now to make sure the right people can have a say in the change.

Done.
Thanks @erikj79 .

**Comment by AlanBateman** [comment]
> Good suggestion!
> And the `jdk_vector_sanity` test group had been added.

In general, running a few fast sanity tests from several areas in tier1 seems a good idea. Having test lists in the TEST.group files isn't very appealing as they easily get out of the sync with the tests in the tree. I realise there are already some test lists in both the hotspot and jdk TEST.groups but it feels like it needs something better so that RunTests.gmk/jtreg can select the sanity tests to run. This is not an objection to this change proposed here, just a comment that this type of configuration is annoying to maintain.

**Comment by mrserb** [comment]
What about possibility to run additional group of the test by passing somehow the name of group to the GA, via label or via /test cmd, or via parameter to the specific task in GA?

**Comment by DamonFool** [comment]
> I realise there are already some test lists in both the hotspot and jdk TEST.groups but it feels like it needs something better so that RunTests.gmk/jtreg can select the sanity tests to run.

Thanks @AlanBateman for this comment.
Is there an existing example to follow?
Thanks.

**Comment by DamonFool** [comment]
> What about possibility to run additional group of the test by passing somehow the name of group to the GA, via label or via /test cmd, or via parameter to the specific task in GA?

Well, it sounds good to run specific tests via label or /test cmd.
However, if the vector api tests are OK to be added in tier1 I think it's fine to run them in GHA directly.

Maybe, we can implement this feature in the future not only for vector api tests but also for other tests too.

**Comment by DamonFool** [comment]
Hi all,

I changed the JBS title as `Add vector api sanity tests in tier1`.
And the added `jdk_vector_sanity` tests can be run in less than 2min on my testing box.
Any comments?

Thanks.

**Review by erikj79 — APPROVED** [review]
This looks good to me. The group tier1_part3 looks like it's currently the fastest of the 3, so it's the right one to add to.

**Review by vnkozlov — APPROVED** [review]
Good.

I think we need review from @shipilev too since he had strong opinion about this.

**Comment by DamonFool** [comment]
Thanks @erikj79 and @vnkozlov for the review.
So @shipilev , are you fine with this change?
Thanks.

File: test/jdk/TEST.groups [hunk 1]
```diff
@@ -75,6 +76,7 @@ tier3 = \
     :jdk_vector \
     :jdk_rmi \
     :jdk_svc \
+   -:jdk_vector_sanity \
    -:jdk_svc_sanity \
```
Comment (shipilev) [inline test/jdk/TEST.groups:80]:
```suggestion
    :jdk_svc \
   -:jdk_svc_sanity \
   -:jdk_vector_sanity \
```

File: test/jdk/TEST.groups [hunk 2]
```diff
@@ -361,6 +363,31 @@ jdk_foreign = \
 jdk_vector = \
     jdk/incubator/vector
 
+jdk_vector_sanity = \
+    jdk/incubator/vector/AddTest.java \
```
Comment (shipilev) [inline test/jdk/TEST.groups:367]: These should probably be sorted too.

File: test/jdk/TEST.groups [hunk 3]
```diff
@@ -39,6 +39,7 @@ tier1_part2 = \
 
 tier1_part3 = \
     :jdk_math \
+    :jdk_vector_sanity \
     :jdk_svc_sanity \
     :jdk_foreign \
```
Comment (shipilev) [inline test/jdk/TEST.groups:44]:
```suggestion
    :jdk_math \
    :jdk_svc_sanity \
    :jdk_foreign \
    :jdk_vector_sanity \
```

**Review by shipilev — COMMENTED** [review]
I think the usual style for these files is to maintain original order/sorting.

**Comment by DamonFool** [comment]
@shipilev , all your comments had been resolved.
Thanks.

**Review by shipilev — APPROVED** [review]

**Comment by DamonFool** [comment]
Thank you all for the review and comments.
/integrate

**Pull request closed without merging**